R/utils.R

```diff
@@ -0,0 +1,222 @@
+#' Detect whether a string contains only numbers
+#'
+#' @param x the string that contains the numbers
+#' @keywords internal
+#' @noRd
+numbers_only <- function(x) {
+  !grepl("\\D", x)
+}
+
+#' Get part1 of date value
+#'
+#' @param x the string of interest
+#' @param sep the separator in the string of interest
+#' @keywords internal
+#' @noRd
+get_part1 <- function(x, sep) {
+  if (is.na(x)) {
+    return(NA)
+  }
+  unlist(strsplit(x, sep, fixed = TRUE))[[1L]]
+}
+
+#' Get part2 of date value
+#'
+#' @param x the string of interest
+#' @param sep the separator in the string of interest
+#' @keywords internal
+#' @noRd
+get_part2 <- function(x, sep) {
+  if (is.na(x)) {
+    return(NA)
+  }
+  unlist(strsplit(x, sep, fixed = TRUE))[[2L]]
+}
+
+#' Get part3 of date value
+#'
+#' @param x the string of interest
+#' @param sep the separator in the string of interest
+#' @keywords internal
+#' @noRd
+#'
+get_part3 <- function(x, sep) {
+  if (is.na(x)) {
+    return(NA)
+  }
+  unlist(strsplit(x, sep, fixed = TRUE))[[3L]]
+}
+
+#' Get sum from number
+#'
+#' @param x the string of interest
+#' @keywords internal
+#' @noRd
+#'
+get_sum <- function(x) {
+  if (nchar(x) == 2L) {
+    x <- sum(as.numeric(substr(x, 1L, 1L)), as.numeric(substr(x, 2L, 2L)))
+  }
+  x
+}
+
+#' Extract letters from a string
+#'
+#' @param x the string of interest
+#' @keywords internal
+#' @noRd
+#'
+only_letters <- function(x) {
+  gsub("^([[:alpha:]]*).*$", "\\1", x)
+}
```

Comment:
Is this intended:

``` r
only_letters <- function(x) {
  gsub("^([[:alpha:]]*).*$", "\\1", x)
}

only_letters("B12A")
#> [1] "B"
```

<sup>Created on 2023-09-28 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

? Or do you expect "BA"?

Reply:
It is actually supposed to return `BA` in your example.

The function is no more used. I will remove it.